### PR TITLE
Feature/non restful sadpath tests

### PR DESCRIPTION
**Pull request by devklynch**
Added tests for sad path and edge cases for the non-restful APIs

Added fix to create new item test

All tests pass

All related non restful API postman tests pass

### CODE REVIEW

File: app/controllers/api/v1/items_controller.rb [hunk 1]
```diff
@@ -52,7 +53,27 @@ def item_params
...
+    def validate_price_params
+        invalid_params = {
+            errors: [
+              {
+                status: "400",
+                message: "Invalid parameters"
+              }
+            ]
+          }
+          if (params[:max_price].present? && params[:max_price].to_f < 0) ||
```
Comment (TDManning) [inline app/controllers/api/v1/items_controller.rb:75]: This is a creative way to test your parameters!

**Review by Paul-Knapp — APPROVED** [review]
Good Job, I like the solution and making sure we can give error messages when appropriate.
